src/TypeScript.UnitTests/Analyzer/EslintBridgeAnalyzerTests.cs

```diff
@@ -284,24 +286,28 @@ public async Task OnSolutionChanged_NextAnalysisCallsInitLinter()
 
             client.Verify(x => x.InitLinter(It.IsAny<IEnumerable<Rule>>(), It.IsAny<CancellationToken>()), Times.Once());
 
-            activeSolutionTracker.Raise(x => x.ActiveSolutionChanged += null, new ActiveSolutionChangedEventArgs(true));
+            activeSolutionTracker.Raise(x => x.ActiveSolutionChanged += null, new ActiveSolutionChangedEventArgs(isSolutionOpen));
 
             await testSubject.Analyze("some path", "some config", CancellationToken.None);
 
-            client.Verify(x => x.InitLinter(It.IsAny<IEnumerable<Rule>>(), It.IsAny<CancellationToken>()), Times.Exactly(2));
+            var expectedCallCount = shouldReinitLinter ? 2 : 1;
+            client.Verify(x => x.InitLinter(It.IsAny<IEnumerable<Rule>>(), It.IsAny<CancellationToken>()), Times.Exactly(expectedCallCount));
         }
 
         [TestMethod]
-        public void OnSolutionChanged_StopsEslintBridgeClient()
+        [DataRow(true, false)]
+        [DataRow(false, true)]
+        public void OnSolutionChanged_StopsEslintBridgeClient(bool isSolutionOpen, bool shouldCallStop)
```

Comment:
The name of the test is no longer correct.
I think for readability, it would be better to split it into 2 tests:
 OnSolutionChanged_NoOpenSolution_StopsEslintBridgeClient
OnSolutionChanged_HasOpenSolution_EslintBridgeClientIsNotStopped (and it's worth aligning it with OnConfigChanged_DoesNotStopEslintBridgeClient)

Reply:
I'll align the test names. I think splitting the tests makes it harder to see the connection between them i.e. that test B is the exact inverse of test A. Using a data driven test makes the connection unmissable.